### PR TITLE
PUBDEV-8099: Fix incorrect Shapley values when there are 0 weights in the tree

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

**Comment by michalkurka** [comment]
@valenad1 this PR has a small API change in the first commit, the main fix is here: https://github.com/h2oai/h2o-3/pull/5568/files#diff-1517845e8dde2b51308e5fc12d8c66156e7ae3f851af2200ee5cb10350b1fe72R153

File: h2o-genmodel/src/test/java/hex/genmodel/algos/tree/TreeSHAPTest.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                    new Node(5, 6, almostZeroWeight * 2, 1, 40),
+                    new Node(regularWeight, 0),
+                    new Node(regularWeight, 0),
+                    new Node(almostZeroWeight, 0),
+                    new Node(almostZeroWeight, 80)
+            };
+        }
+    }
+
+    private static class Model_Appp_TreeBuilder implements TreeBuilder { // Model A''' from support ticket #99203 (private)
```
Comment (valenad1) [inline h2o-genmodel/src/test/java/hex/genmodel/algos/tree/TreeSHAPTest.java:52]: What is the difference between Model_App and Model_Appp, what is means?

**Review by valenad1 — APPROVED** [review]
LGTM, thank you!

File: h2o-genmodel/src/test/java/hex/genmodel/algos/tree/TreeSHAPTest.java (hunk 1 above)
Comment (michalkurka) [reply h2o-genmodel/src/test/java/hex/genmodel/algos/tree/TreeSHAPTest.java:52]: thank you for this comment - I will address it in a new PR

**Pull request merged**